test_conformance/subgroups/subgroup_common_templates.h

```diff
@@ -388,11 +390,57 @@ template <typename Ty> bool is_floating_point()
         || std::is_same<Ty, subgroups::cl_half>::value;
 }
 
+// limit possible input values to avoid arithmetic rounding/overflow issues.
+// for each subgroup values defined different values
+// for rest of workitems set 1
+// shuffle values
+template <typename Ty>
+void fill_and_shuffle_safe_values(std::vector<cl_ulong> &safe_values,
+                                  const int &sb_size)
+{
+    // below values are 0.5f, 0.25f, 2.0f, 4.0f
+    std::vector<cl_ulong> safe_half_values{ 0x3800, 0x3400, 0x4000, 0x4400 };
+    // max possible value is == 5040
+    std::vector<cl_ulong> safe_other_types_values{ 2, 3, 4, 5, 6, 7 };
+
+    cl_ulong set_one = 0x3c00;
+
+    if (std::is_same<Ty, subgroups::cl_half>::value)
+    {
+        safe_values = safe_half_values;
+    }
+    else
+    {
+        safe_values = safe_other_types_values;
+        set_one = 1;
+    }
+
+    int offset_size = sb_size - safe_values.size();
+    if (offset_size > 0)
+    {
+        safe_values.insert(safe_values.end(), offset_size, set_one);
+    }
+    else
+    {
+        safe_values.resize(sb_size);
```

Comment:
Does this mean that for a subgroup size less than four, not all of the values in safe_half_values will get used, and for a subgroup size less than six, not all of the values in safe_other_type_values will get used.  Is this really appropriate?

Reply:
now all types take part from non_one_values

Reply:
The same question is now equally valid for non_one_values, though!

@jlewis-austin, do you have any thoughts or opinions about this?  I cannot see a good reason that the numerical coverage for small subgroup sizes should be less than that for large subgroup sizes, but perhaps I've missed something in Grzegorz's code?

Reply:
I am not sure where is the problem. I need to prepare values for particular subgroup size. There is no ability to set more values that subgroup size that why this limitation.

Reply:
> Does this mean that for a subgroup size less than four, not all of the values in safe_half_values will get used, and for a subgroup size less than six, not all of the values in safe_other_type_values will get used.

Yes (now with `non_one_values`).

> Is this really appropriate?

Yes, why not? As @gwawiork noted, we cannot have more values than the subgroup size. And we cannot have too many big numbers in order to prevent multiplication overflows.

Reply:
Also, the whole point of these "magical values" is that they generate no multiplication (or addition) overflow. Testing with the small ones is just as good as the larger ones.

Reply:
> The same question is now equally valid for non_one_values, though!
> 
> @jlewis-austin, do you have any thoughts or opinions about this? I cannot see a good reason that the numerical coverage for small subgroup sizes should be less than that for large subgroup sizes, but perhaps I've missed something in Grzegorz's code?

@StuartDBrady I still have some questions, but haven't had time to dig deep enough to understand the big picture. It seems like coverage would benefit from having more unique values involved, but I'm OK with this letting this go in while we chew on it.